src/resolve-url.js

```diff
@@ -0,0 +1,17 @@
+import url from 'url';
+import defaults from 'lodash.defaults';
+import pickBy from 'lodash.pickby';
+
+const resolveUrl = function(urlString, query) {
+  const parsedUrl = url.parse(urlString, true);
+
+  const mergedQuery = defaults({}, query, parsedUrl.query);
+  const fullQuery = pickBy(mergedQuery, (value) => value);
+
+  delete parsedUrl.search;
```

Comment:
Is this actually necessary? I want to say that `url` is smart enough to prefer `query` to search, but I might be wrong.